### PR TITLE
Uniform pilot-agent and pilot-discover 's code style

### PR DESCRIPTION
**Pull request by vince-0202**
**Please provide a description of this PR:**
I noticed that pilot-agent's code style was different from pilot-discovery's,I wonder if it would be better like that.

I uniformed pilot-agent and pilot-discover's code style in this PR #38490 ,and suggested that I should fix it in master branch.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -78,7 +78,7 @@ require (
 	gomodules.xyz/jsonpatch/v3 v3.0.1
 	google.golang.org/api v0.74.0
 	google.golang.org/genproto v0.0.0-20220414192740-2d67ff6cf2b4
-	google.golang.org/grpc v1.45.0
+	google.golang.org/grpc v1.46.0
```
Comment (hzxuzhonghu) [inline go.mod:81]: Donot couple this

**Review by hzxuzhonghu — COMMENTED** [review]
I like the way removing the long init function. Just a little worried about we are lack of tests

**Comment by vince-0202** [comment]
> I like the way removing the long init function. Just a little worried about we are lack of tests

I have the same concern,It's my first PR for Istio.Is there anything I can do to help with the testing

**Pull request closed without merging**